crypto/Cargo.toml

```diff
@@ -23,13 +23,17 @@ harness = false
 required-features = ["concurrent"]
 
 [features]
+alloc = ["hashbrown", "math/alloc", "utils/alloc"]
+default = ["std"]
 concurrent = ["rayon"]
+std = ["blake3/std", "math/std", "utils/std", "sha3/std"]
 
 [dependencies]
-utils = { path = "../utils", package = "winter-utils" }
-math = { path = "../math", package = "winter-math" }
-blake3 = "1.0"
-sha3 = "0.9"
+utils = { path = "../utils", package = "winter-utils", default-features = false }
+math = { path = "../math", package = "winter-math", default-features = false }
+blake3 = { version = "1.0", default-features = false }
+sha3 = { version = "0.9", default-features = false }
+hashbrown = { version = "0.9.1", optional = true, default-features = false, features = ["ahash", "inline-more"] }
```

Comment:
I wonder if we should move `hashbrown` dependency to the `utils` create. Then, in the `utils` crate we could create `collections` module which could look something like this:

```Rust
#[cfg(feature = "alloc")]
pub use alloc::collections::{BTreeMap, BTreeSet};

#[cfg(feature = "alloc")]
pub use alloc::vec::Vec;

#[cfg(feature = "alloc")]
pub use hashbrown::HashMap;

#[cfg(feature = "std")]
pub use std::collections::{BTreeSet, HashMap};

#[cfg(feature = "std")]
pub use std::vec::Vec;
```

Then, in many places when we need to import these structs, we could just do `use utils::collections::*;`

Reply:
Will be cleaner indeed. Will do!